### DISCUSSION TITLE
Grafana container keeps stopping unexpectedly after 1 - 2 minutes

### DISCUSSION THREAD

**Discussion by jeffkruczynski** [jasonacox/Powerwall-Dashboard#122] [Q&A] [answered]
Hello,
(Running on Synology NAS)
I am running into an issue where the Grafana container keeps stopping unexpectedly. In the log file, it seems to stop immediately after an error:
Error: Get "https://grafana.com/api/plugins/repo/grafana-piechart-panel": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

I had previously modified the `powerwall.yml` file to set Grafana to user 0:0 because I was running into permissions errors for the various /var/lib/grafana locations. Changing the user seems to have resolved those, but now I can't get past this "deadline exceeded" error.

I thought this might be a firewall issue; I'm able to access that URL directly from the browser. I do have 9000 open on the NAS and on the router, but I don't know if that's relevant here — I'm guessing this is just using http:?
(I did notice that the URL is https: not http:, so I uncommented the lines in `grafana.env` in the https section. That didn't fix it.

I'm a bit over my head here — can anyone point me in a direction to try to resolve this?
Thank you!

**Comment by jeffkruczynski** [comment]
@jasonacox in browsing for solutions to this, I found [this thread](https://github.com/geerlingguy/internet-pi/issues/349) which seems to imply setting the DNS when launching the grafana container might be solution. However, I can't find where I would adjust this in Powerwall-Dashboard. Is there a way here to specify the DNS when starting grafana within this project?
(and please let me know if I'm completely on the wrong path here...)

**Comment by jeffkruczynski** [comment, marked as answer]
Finally figured this out...turned out I needed to create a NAS firewall rule for the Docker subnets. Now the container runs fine, and I just need to figure out how to get to it via the web interface.

**Reply by jasonacox** [reply]
Thanks for opening this @jeffkruczynski !  Good discovery.

When grafana starts up, it attempts to download all the plugins from grafana.com which is why you saw the error trying to pull down that URL.  It sounds like you figured out to allow it to call the https://grafana.com resources.  I would like to add whatever steps you had to take to make that work for others trying to get it to work.

The dashboard project communicates and listens on these TCP ports:
* 8086 - InfluxDB
* 8675 - pyPowerwall - you can see a PW stats page at http://localhost:8675/stats or vitals at http://localhost:8675/vitals
* 8676 - weather411 - current conditions at http://localhost:8676/
* 9000 - Grafana - this is the entry point for the dashboard (e.g. http://localhost:9000) 

I love to get a Synology to help troubleshoot or make the setup easier (hint, hint, Santa :-) but I'm happy to add any tips/tricks you discovered to our [instructions](https://github.com/jasonacox/Powerwall-Dashboard#synology-nas-and-rootless-docker).  What Synology model are you using?

**Reply by jeffkruczynski** [reply]
I was on the wrong path first because I figured the grafana setup needed to be fixed within pypowerwall. Once I figured out it that wasn't the case (I tested for this by installing the grafana docker image directly from the registry in Synology — it couldn't reach the plugin URLs for installation either), then I needed to figure out how that access was being blocked.

[This post](https://community.synology.com/enu/forum/11/post/142221) helped me figure out what needed to be done: ensure that the docker network could get through the NAS firewall. But I didn't know what firewall entry to make, and I didn't understand the network structure for multiple docker containers communicating with each other. [This page](https://www.freecodecamp.org/news/how-to-get-a-docker-container-ip-address-explained-with-examples/) helped me understand how the network(s) are structured and how to get the IP addresses needed to create the firewall entry.

Also, in order to get the IP address for the grafana container, I had to remove the GF_INSTALL_PLUGINS environment variable so that the container would stay running long enough for me to get the address. Once I had the IP address, I added that variable back in; then, I created the firewall entry as shown in the forum link above, and it worked.

So:
- Removed the GF_INSTALL_PLUGINS environment variable & restarted the container
- Figure out the subnet IP address for the grafana container that needed access through the firewall
- Added a firewall entry to allow that subnet access
- Stopped the container & re-added the GF_INSTALL_PLUGINS environment variable.

(Note: I just gave the subnet access to all ports. It could be limited to just those four you listed, I suppose.)

I'm running this on a DS916+, DSM 7.1.
